### PR TITLE
Bug fix blog post link

### PR DESCRIPTION
**Pull request by titame**
Hello @rohandaxini ,

This PR contains changes for using hex url for user post link. Please review and let me know if any changes are required.

Thanks,
Tushar TItame

### CODE REVIEW

**Comment by rohandaxini** [comment]
@titame ill start reviewing this PR now and planning to merge it soon. Lets discuss.

**Comment by titame** [comment]
Sure @rohandaxini . We can discuss and merge it.

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -119,6 +119,18 @@ def search
       params[:search].present?
   end
 
+  def hex_url_to_frindly_url
```
Comment (rohandaxini) [inline app/controllers/posts_controller.rb]: @titame typo here, it should be `hex_url_to_friendly_url` check spelling if friendly

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -17,7 +17,8 @@ def timeago_date_format(data)
   end
 
   def post_link(post)
-    link_to truncate(post.subject, length: 70), post_path(post)
+    link_to truncate(post.subject, length: 70),
+      shared_post_path(url_token: post.url_token)
```
Comment (titame) [inline app/helpers/application_helper.rb]: Sure @rohandaxini .

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (titame) [inline app/controllers/posts_controller.rb]: Sure @rohandaxini .

File: lib/tasks/set_url_token_for_posts.rake [hunk 3]
```diff
@@ -0,0 +1,21 @@
+namespace :db do
+  desc "To set url token for existing posts"
+  task set_url_token_for_posts: :environment do
+    begin
+      count = 0
+      Post.find_each(batch_size: 10) do |post|
+        next if(post.url_token.present?)
+        puts "=== generating url token for post id: ##{post.id} ==="
```
Comment (rohandaxini) [inline lib/tasks/set_url_token_for_posts.rake]: @titame mention Post title instead of ID as ID won't make it clear which blog post is getting updated.

Comment (titame) [inline lib/tasks/set_url_token_for_posts.rake]: Sure @rohandaxini .

**Pull request merged**